contrib/golang/common/dso/dso.cc

```diff
@@ -53,9 +26,18 @@ DsoInstance::DsoInstance(const std::string dso_name) : dso_name_(dso_name) {
   handler_ = dlopen(dso_name.c_str(), RTLD_LAZY);
   if (!handler_) {
     ENVOY_LOG_MISC(error, "cannot load : {} error: {}", dso_name, dlerror());
-    return;
   }
+}
+
+DsoInstance::~DsoInstance() {
+  // TODO(doujiang24): make sure the dso file is not no other users,
```

Comment:
shared_ptr is used, so i don't think it's a issue.

Reply:
yep, you are right, also added comments for it, thanks.